apps/api/src/app/controllers/resources/term.controller.ts

```diff
@@ -37,6 +47,26 @@ export interface UserQueryOptions {
     // TODO[https://coscrad.atlassian.net/browse/CWEBJIRA-328] Support custom user-defined sort order
 }
 
+const extractPathsFromUserFilter = (filter: CoscradFilterCondition, paths = []): string[] => {
+    const { type } = filter;
+
+    if (type === CoscradConditionBlockType.SIMPLE) {
+        paths.push((filter as CoscradSimpleCondition).field);
+
+        return paths;
+    }
+
+    if (type === CoscradConditionBlockType.AND || type === CoscradConditionBlockType.OR) {
+        (filter as CoscradAndCondition).conditions.forEach((condition) => {
+            paths.push(...extractPathsFromUserFilter(condition));
+        });
+
+        return paths;
+    }
+
```

Comment:
Just want to note that this will blow up when we have the occasion to use a not filter. We don't need it right now, though.